src/components/AttachmentCarousel/index.js

```diff
@@ -108,12 +112,42 @@ class AttachmentCarousel extends React.Component {
         });
     }
 
+    /**
+     * On a touch screen device, automatically hide the arrows
+     * if there is no interaction for 3 seconds.
+     */
+    autoHideArrow() {
+        if (!this.canUseTouchScreen) {
+            return;
+        }
+        this.cancelAutoHideArrow();
+        this.autoHideArrowTimeout = setTimeout(() => {
+            this.toggleArrowsVisibility(false);
+        }, CONST.ARROW_HIDE_DELAY);
+    }
+
+    /**
+     * Cancels the automatic hiding of the arrows.
+     */
+    cancelAutoHideArrow() {
+        clearTimeout(this.autoHideArrowTimeout);
+    }
+
     /**
      * Toggles the visibility of the arrows
      * @param {Boolean} shouldShowArrow
      */
     toggleArrowsVisibility(shouldShowArrow) {
-        this.setState({shouldShowArrow});
+        this.setState((current) => {
+            const newShouldShowArrow = !_.isUndefined(shouldShowArrow) ? shouldShowArrow : !current.shouldShowArrow;
+            return {shouldShowArrow: newShouldShowArrow};
+        }, () => {
```

Comment:
In `renderCell`, there is a `Pressable` which will toggle the value based on the previous value.

Reply:
We call the auto hide arrow inside `toggleArrowsVisibility`. Because we set the state manually (`this.setState(current => ({shouldShowArrow: !current.shouldShowArrow}))`), the arrow never auto hide and I think we will be repeating the logic to show/hide the arrow inside the `setState` callback

Reply:
Yes, that's why if we pass nothing to `toggleArrowVisibility`, we set the value based on the previous state.